pkg/controller/trigger_controller.go

```diff
@@ -126,50 +132,30 @@ func NewTriggerController(
 	return controller
 }
 
-func indexDeploymentsByTriggeringConfigMaps(obj interface{}) ([]string, error) {
-	var configs []string
-	d, ok := obj.(*appsv1.Deployment)
-	if !ok {
-		return configs, fmt.Errorf("object is not deployment")
-	}
-	annotations := d.GetAnnotations()
-	if len(annotations) == 0 {
-		return configs, nil
+func indexByConfigMaps(obj interface{}) ([]string, error) {
+	objMeta, err := meta.Accessor(obj)
+	if err != nil {
+		return []string{}, nil
 	}
+
+	annotations := objMeta.GetAnnotations()
 	if triggers, ok := annotations[triggerConfigMapsAnnotation]; ok {
-		configMaps := sets.NewString(strings.Split(triggers, ",")...)
-		for _, v := range d.Spec.Template.Spec.Volumes {
```

Comment:
The original behavior, whereby it cross-referenced volumes, was a bit unexpected to me. If I'm adding the `triggering` annotations, I expect rollouts to occur regardless of how the secret is being used.

E.g. I might be:
- Using e.g. `configMapKeyRefs` in environment variables.
- Using volumes (as shown).

Because folks have to opt-in to this behavior, I believe it's fairly explicit to state that we'll rollout on any change to a specific secret or configMap. I'd be reticent to make it any more complex and say, configure the conditions that cause us to trigger via additional annotations.

---

Rather than taking the intersection of the annotation values the stated volumes in our Spec, I think it would be interesting to automatically trigger on any referenced secrets or configmaps, by they envKeyRefs, or volumes. _That_ would make sense as an explicit annotation e.g. `trigger.k8s.io/auto: enabled`